### PR TITLE
Task 6 implementation, tests, endpoint description, setup

### PR DESCRIPTION
**Pull request by serhii-fsi**
# Task 6 - CORE: GET /api/articles/:article_id/comments
- [x] "Jest-sorted" setup
- Integration tests for endpoint `GET /api/articles/:article_id/comments`
    - [x] Returns all comments for the 1st article with the correct prop types,
        checks comments number
    - [x] Comments should be served with the most recent comments first
    - [x] Fields of the 2nd comment should be the same
    - [x] Returns an empty array for the 2nd article which doesn't have comments 
    - [x] Returns 404 error for incorrect but valid article_id
    - [x] Returns 400 error for invalid article_id 
- Implementation
    - [x] Route
    - [x] Controller
    - [x] Model
- [x] Tests fixes
- [x] Passing tests
- [x] Add a description of this endpoint

### CODE REVIEW

File: tests/integration/api-v1/articles.test.js [hunk 1]
```diff
@@ -149,16 +149,88 @@ describe("GET /api/articles", () => {
...
+
+
+describe("GET /api/articles/:article_id/comments", () => {
+
+    test("returns all comments for the 1st article with the correct prop types", () => {
+        return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then(({ body: { comments } }) => {
+                expect(comments).toHaveLength(11);
```
Comment (nc-rob-anderson) [inline tests/integration/api-v1/articles.test.js:166]: Well done for including the length assertion :)

File: tests/integration/api-v1/articles.test.js [hunk 2]
```diff
@@ -149,16 +149,88 @@ describe("GET /api/articles", () => {
...
+    });
+
+    test("returns an empty array for the 2nd article which doesn't have comments", () => {
+        return request(app)
+            .get('/api/articles/2/comments')
+            .expect(200)
+            .then(({ body: { comments } }) => {
+                expect(comments).toEqual([]);
+            });
+    });
```
Comment (nc-rob-anderson) [inline tests/integration/api-v1/articles.test.js:214]: This is an important test to include, well done :)

File: tests/integration/api-v1/articles.test.js [hunk 3]
```diff
@@ -149,16 +149,88 @@ describe("GET /api/articles", () => {
...
+        return request(app)
+            .get('/api/articles/invalid_article_id/comments')
+            .expect(400)
+            .then(({ body: { error } }) => {
+                const appErr = new AppError({ code: 400, msg: '400 Bad Request', log: false });
+                expect({ error }).toMatchObject(appErr.exportForClient());
+            });
+    });
+
+});
```
Comment (nc-rob-anderson) [inline tests/integration/api-v1/articles.test.js:236]: Great sad path tests also

File: src/models/fetch-article-existence.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            FROM articles
+            WHERE article_id = $1;`,
+            [articleId]
+        );
+        return rows.length === 1;
+    } catch (error) {
+        const modelErr = new ModelError({ psql: error, msg: 'PSQL Error' });
+        throw modelErr.toAppError();
+    }
+};
```
Comment (nc-rob-anderson) [inline src/models/fetch-article-existence.js:19]: Just a thought, You could argue this is a very similar function to fetchArticleById, and that could be used instead

Comment (nc-rob-anderson) [reply src/models/fetch-article-existence.js:19]: But it is clearer in what function it's doing, so it makes sense

**Review by nc-rob-anderson — COMMENTED** [review]
Great work Serhii

File: src/models/fetch-article-existence.js (hunk 4 above)
Comment (serhii-fsi) [reply src/models/fetch-article-existence.js:19]: You are absolutely right. I forgot that I have the same) Thank you!

**Pull request merged**